src/generic-container.ts

```diff
@@ -36,6 +36,9 @@ export class GenericContainer implements TestContainer {
   private cmd: Command[] = [];
   private waitStrategy?: WaitStrategy;
   private startupTimeout: Duration = new Duration(60_000, TemporalUnit.MILLISECONDS);
+  private name?: string;
```

Comment:
We're introducing types to make them easier to validate/refactor in the future. 
Eg `name: ContainerName`
Could we do the same here?